Lib/encodings/idna.py

```diff
@@ -186,13 +202,22 @@ def encode(self, input, errors='strict'):
             if result:
                 # Join with U+002E
                 result.extend(b'.')
-            result.extend(ToASCII(label))
+            try:
+                result.extend(ToASCII(label))
+            except UnicodeEncodeError as exc:
+                raise UnicodeEncodeError(
+                    "idna",
+                    input,
+                    len(result) + exc.start,
+                    len(result) + exc.end,
+                    exc.reason,
+                )
```

Comment:
`len(result)` is not a good offset.

```
>>> "あさ".encode("idna")
b'xn--l8jvb'

>>> "あさ.\u034f".encode("idna")
Traceback (most recent call last):
  File "/Users/inada-n/work/python/cpython/Lib/encodings/idna.py", line 206, in encode
    result.extend(ToASCII(label))
                  ~~~~~~~^^^^^^^
  File "/Users/inada-n/work/python/cpython/Lib/encodings/idna.py", line 93, in ToASCII
    raise UnicodeEncodeError("idna", label, 0, len(label), "label empty or too long")
UnicodeEncodeError: 'idna' codec can't encode characters in position 0--1: label empty or too long

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    "あさ.\u034f".encode("idna")
    ~~~~~~~~~~~~~~~~~~~~^^^^^^^^
  File "/Users/inada-n/work/python/cpython/Lib/encodings/idna.py", line 208, in encode
    raise UnicodeEncodeError(
    ...<5 lines>...
    )
UnicodeEncodeError: 'idna' codec can't encode characters in position 10-9: label empty or too long
encoding with 'idna' codec failed
```

Reply:
Good call, fixed this to use the lengths of the input seen so far rather than the result string.